src/support/json.h

```diff
@@ -272,12 +272,17 @@ struct Value {
       // String
       // Start |close| at the opening ", and in the loop below we will always
       // begin looking at the first character after.
-      char* close = curr;
-      // Skip escaped "
-      do {
-        close = strchr(close + 1, '"');
-      } while (*(close - 1) == '\\');
-      assert(close);
+      char* close = curr + 1;
+      // Skip escaped ", which appears as \". We need to be careful though, as
+      // \" might also be \\" which would be an escaped \ and an *un*escaped ".
+      while (*close && *close != '"') {
+        if (*close == '\\') {
+          // Skip the \ and the character after it, which it escapes.
+          close++;
+          assert(*close);
+        }
+        close++;
```

Comment:
Nit: `++close` here and above.

Reply:
These are just pointers, and the current code is consistent with usage elsewhere in the file - is it worth adding an inconsistency here do you think?

Reply:
No, probably not.